tests/js/client/shell/shell-move-shard-sync-fail-cluster.js

```diff
@@ -0,0 +1,197 @@
+        // are supposed to swap during the moveShard!
+        // We want that first the old leader discovers a SynchronizeShard,
+        // and that then the new leader takes over shard leadership,
+        // tells its followers and reports to Current, and that **only then**
+        // the SynchrnoizeShard job is executed on the old leader (now 
+        // follower) and then fails.
+        let prog = [
+          `SynchronizeShard::beginning ${collInfo.leader}:${collInfo.shard} SynchronizeShardStarted`,
+          `HandleLeadership::before ${collInfo.follower}:${collInfo.shard} HandleLeadershipBefore`,
+          `HandleLeadership::after ${collInfo.follower}:${collInfo.shard} HandleLeadershipAfter`,
+          `Maintenance::BeforePhaseTwo ${collInfo.follower} LeaderSendsCurrent`,
+          `Maintenance::AfterPhaseTwo ${collInfo.follower} LeaderSentCurrent`,
+          `ClusterInfo::loadCurrentSeesLeader ${collInfo.leader}:${collInfo.shard}:${followerId} FollowerUpdatesCurrent`,
+          `ClusterInfo::loadCurrentDone ${collInfo.leader} FollowerHasUpdatedCurrent`,
+          `SynchronizeShard::beginning2 ${collInfo.leader}:${collInfo.shard} SynchronizeShardStartedContinuing`,
+          `SynchronizeShard::beforeSetTheLeader ${collInfo.leader}:${collInfo.shard} SynchronizeShardRunning`,
+          `SynchronizeShard::beginning ${collInfo.leader}:${collInfo.shard} SynchronizeShardStarted2`,
+          `SynchronizeShard::beginning2 ${collInfo.leader}:${collInfo.shard} SynchronizeShardStartedContinuing2`,
+          `SynchronizeShard::beforeSetTheLeader ${collInfo.leader}:${collInfo.shard} SynchronizeShardRunning2`,
+          ""  // want to have a new line at the end of the last line!
+        ].join("\n");
+        fs.writeFileSync(progPath, prog);
+        fs.writeFileSync(pcPath, "");
+
+        // Activate failure points in leader and follower:
+        debugSetFailAt(followerEndpoint, "HandleLeadership::before");
+        debugSetFailAt(followerEndpoint, "HandleLeadership::after");
+        debugSetFailAt(followerEndpoint, "Maintenance::BeforePhaseTwo");
+        debugSetFailAt(followerEndpoint, "Maintenance::AfterPhaseTwo");
+        debugSetFailAt(leaderEndpoint, "SynchronizeShard::beginning");
+        debugSetFailAt(leaderEndpoint, "SynchronizeShard::beforeSetTheLeader");
+        debugSetFailAt(leaderEndpoint, "ClusterInfo::loadCurrentSeesLeader");
+        debugSetFailAt(leaderEndpoint, "ClusterInfo::loadCurrentDone");
+        debugSetFailAt(leaderEndpoint, "SynchronizeShard::fail");
+
+        // Now move the shard:
+        let res = moveShard("_system", cn, collInfo.shard, collInfo.leader, collInfo.follower, false /* dontwait */);
+
+        // See if the program was executed:
+        let count = 0;
+        let nrSteps;
+        while (count <= 600) {
+          let steps = fs.readFileSync(pcPath).toString().split("\n");
+          nrSteps = steps.length - 1;
+          if (nrSteps === 12) {
```

Comment:
Probably better to do it like this, so there is a visible relationship with `prog` for the reader of the test code.
```suggestion
          if (nrSteps === prog.length - 1) {
```

Reply:
Done, needed to do a small change to the definition of prog for this to work.